assignments/005-slurm-and-bwa-mem2/map2.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+#SBATCH --cpus-per-task=1
+#SBATCH --time=1:00:00
+#SBATCH --output=bwa_map_array2-%A_%a.out
+#SBATCH --error=bwa_map_array2-%A_%a.err
+#SBATCH --array=1-16
+
+# load the module that gives us the bwa software
+#module load bwa
+module load samtools
+
+# make a directory for log output if it does not already exist
+LDIR=results/log/bwa_map_array2
+ODIR=results/mapped2
+
+mkdir -p $LDIR $ODIR
+
+# get shell variables for this array task:
+COMM=$(./scripts/line-assign.sh $SLURM_ARRAY_TASK_ID data/sample-array-info2.tsv)
+eval $COMM
+
+
+# run bwa mem on the input and pipe it to samtools to sort it
+./../miniforge3/bin/bwa-mem2 mem \
+  -R "@RG\tID:${sample}.${library}.${flowcell}.${lane}\tSM:${sample}\tLB:${library}\tBC:${barcode}\tPU:\tID:${sample}.${library}.${flowcell}.${lane}.${barcode}\tPL:ILLUMINA" \
+  data/genome/genome.fasta \
+  $fq1 $fq2  2> $LDIR/bwa_mem_$sample.log | \
```

Comment:
This is using the raw fastqs.  We want to map the _trimmed_ fastqs.  These are in results/trimmed/*.fq.gz.

You would replace $fq1 and $fq2 with

```
results/trimmed2/${sample}_R1.fq.gz 
  results/trimmed2/${sample}_R2.fq.gz
```
the $sample variable is obtained from the line-assign.sh script and `eval $COMM`